File: Frontend/GraduationProject/src/app/models/offer.ts
import { Bill } from './Bill';
import { Customer } from './Customer';
import { Product } from './product';

export interface Offer {
  offerNumber: number;

  billId: number | undefined;

  offerStatus: OfferStatus;

  endTime: Date | undefined;

  bill: Bill | undefined;

  customer: Customer | undefined;

  selectedProduct: Product | undefined;

  offerDetails: Array<OfferDetail>;
}

export interface OfferDetail {
  offerId: number;

  customerId: string;

  height: number;

  weight: number;

  relationship: Relationships;

  job: string;

  customer: Customer;
}

export enum OfferStatus {
  insuredsAdding,

  productSelection,

  installmentSelection,

  paymentPending,

  completed,

  insurerAdding,
}

export enum Relationships {
  itSelf,

  partner,

  child,
}

export class AddOfferModel {
  constructor(public customerId: string) {}
}

export class AddOfferDetailModel {
  constructor(
    public customerId: string,
    public height: number,
    public weight: number,
    public relationship: Relationships,
    public job: string
  ) {}
}

export class OfferUpdateProduct {
  constructor(
    public offerNumber: number,
    public selectedProduct: number,
    public offerStatus: OfferStatus
  ) {}
}

File: Backend/Graduation.Project/Graduation.Project.Entity/Models/Offer.cs
using Graduation.Project.Entity.Base;
using Graduation.Project.Entity.Static;
using System;
using System.Collections.Generic;

namespace Graduation.Project.Entity.Models
{
    public class Offer : EntityBase
    {
        public Offer()
        {
            OfferDetails = new HashSet<OfferDetail>();
        }

        public decimal OfferNumber { get; set; }
        public decimal? SelectedProduct { get; set; }
        public decimal? BillId { get; set; }
        public OfferStatus OfferStatus { get; set; }
        public string CustomerId { get; set; } = null!;
        public DateTime? EndTime { get; set; }

        public virtual Bill? Bill { get; set; }
        public virtual Customer Customer { get; set; } = null!;
        public virtual Product? SelectedProductNavigation { get; set; }
        public virtual ICollection<OfferDetail> OfferDetails { get; set; }
    }
}
